### PR TITLE
feat/completion-pum

### LINKED ISSUE #357: feat: add completion box
see #355

### PR DESCRIPTION
**Pull request by mosheavni**
fixes #357 

WORK IN PROGRESS
image

### CODE REVIEW

**Comment by mosheavni** [comment]
currently:


issues:

- [ ] long results are currently not shown (more than 50)
- [ ] pum stays visible for a few seconds after the selection is chosen
- [ ] vim.on_key is on every buffer. how can I limit to only the prompt buffer (alias/ns/ctx)?

@Ramilito  need ur help here 🙏🏼
you'd probably also want to move stuff around

**Comment by Ramilito** [comment]
The gifs are looking soooo goood!
I'll try to get some time later tonight and look if I can help

**Comment by mosheavni** [comment]
> The gifs are looking soooo goood!
> I'll try to get some time later tonight and look if I can help

Please work your magic here because I am really stuck and you seem to get the hang of all those UI components

**Comment by Ramilito** [comment]
>  long results are currently not shown (more than 50)
 pum stays visible for a few seconds after the selection is chosen
 vim.on_key is on every buffer. how can I limit to only the prompt buffer (alias/ns/ctx)?

- Long results, I removed the limit you added and added a windows height restriction instead, so we would scroll for the rest.
- Not seeing the delay, is it still there?
- vim.on key wasn't called because the buffer wasn't really removed (just hidden), I set it to wipe on close and now it's getting called

Can you verify that I understood this correctly?

**Comment by mosheavni** [comment]
yes, it looks great :)
I changed row to 1 to move the pum a line lower so it won't hide the text, and changed some stuff around the highlights. 
ok to merge? :)

File: lua/kubectl/actions/highlight.lua [hunk 1]
```diff
@@ -31,12 +31,13 @@ local highlights = {
...
   KubectlNote = { fg = "#9CDCFE" }, -- Light Blue
   KubectlGray = { fg = "#666666" }, -- Dark Gray
+  KubectlPum = { bg = "#3e4451" },
   KubectlUnderline = { underline = true },
 }
 
 function M.setup()
   for group, attrs in pairs(highlights) do
-    local success, hl = pcall(api.nvim_get_hl_by_name, group, true)
+    local success, hl = pcall(api.nvim_get_hl, 0, group)
```
Comment (Ramilito) [inline lua/kubectl/actions/highlight.lua:40]: nice!

**Review by Ramilito — APPROVED** [review]

**Comment by Ramilito** [comment]
It looks sooo cool!
Love your work here!

**Pull request merged**